src/Symfony/Bundle/TwigBundle/Cache/ChainCache.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\Cache;
+
+use Twig\Cache\CacheInterface;
+
+/**
+ * Chains several caches together.
+ *
+ * Cached items are fetched from the first cache having them in its data store.
+ * They are saved and deleted in all adapters at once.
+ */
+class ChainCache implements CacheInterface
```

Comment:
Would it make sense to contribute this class to Twig instead?

Reply:
I was afraid that the dependency upgrade bump would be too big. Would it be acceptable ? I see the bundle requires twig `^3.0.4` for now, latest twig version is `3.8.0`.

Reply:
You can make this class `@internal` and contribute it to twig. We will then be easier to delete this class when we switch to the Twig class.